lighthouse-cli/bin.ts

```diff
@@ -287,34 +253,44 @@ function handleError(err: LightHouseError) {
   }
 }
 
-function run() {
-  return initPort(cliFlags).then(() => {
-    if (cliFlags.skipAutolaunch) {
-      return lighthouseRun(urls, config, cliFlags).catch(handleError);
-    } else {
-      // because you can't cancel a promise yet
-      const isSigint = new Promise((resolve, reject) => {
-        process.on(_SIGINT, () => reject(_SIGINT));
-      });
-
-      return Promise
-        .race([launchChromeAndRun(urls, config, cliFlags), isSigint])
-        .catch(maybeSigint => {
-          if (maybeSigint === _SIGINT) {
-            return cleanup
-              .doCleanup()
-              .catch(err => {
-                console.error(err);
-                console.error(err.stack);
-              }).then(() => process.exit(_ERROR_EXIT_CODE));
-          }
-          return handleError(maybeSigint);
+function runLighthouse(url: string,
+                       flags: {port: number, skipAutolaunch: boolean, selectChrome: boolean, output: any, outputPath: string, view: boolean},
+                       config: Object): Promise<undefined> {
+
+  let chromeLauncher: ChromeLauncher;
+  return initPort(flags)
+    .then(() => getDebuggableChrome(flags))
+    .then(chrome => chromeLauncher = chrome)
```

Comment:
just tried it and indentation becomes two levels deeper and it's a little more work to follow (whereas having it flat like this makes the steps very obvious, and every step essentially has a clean slate (except the `chromeLauncher` var sticking around)). It may be worth it if we need to save more state than just the chrome instance, but in this case I think it's better this way.